master/internal/rm/kubernetesrm/jobs.go

```diff
@@ -1850,6 +1856,25 @@ func (j *jobsService) verifyNamespaceExists(namespaceName string) error {
 	return nil
 }
 
+func (j *jobsService) removeEmptyNamespace(namespaceName string, clusterName string) error {
+	count, err := workspace.GetNumWorkspacesUsingNamespaceInCluster(context.Background(), clusterName, namespaceName)
+	if err != nil {
+		return err
+	}
+	if count == 0 {
+		delete(j.podInterfaces, namespaceName)
+		delete(j.configMapInterfaces, namespaceName)
+		delete(j.jobInterfaces, namespaceName)
+
+		for _, worker := range j.requestQueueWorkers {
+			worker.podInterface = j.podInterfaces
+			worker.configMapInterfaces = j.configMapInterfaces
+			worker.jobInterface = j.jobInterfaces
+		}
+	}
```

Comment:
What happens in the API if we can't remove the namespace because it's not empty? Does the API not care or does it need to know?

Reply:
It does not care